server/api.go

```diff
@@ -289,6 +293,16 @@ func (p *Plugin) downloadLegalHold(w http.ResponseWriter, r *http.Request) {
 	}
 }
 
+func (p *Plugin) runJobFromAPI(w http.ResponseWriter, _ *http.Request) {
```

Comment:
I don't think this will work where a long running job is initiated and the HTTP request waits for it to complete.  That will timeout.  Also we need to make sure triggering the job manually cannot lead to having two jobs running.

Reply:
> I don't think this will work where a long running job is initiated and the HTTP request waits for it to complete. That will timeout.

The response is now written before the job runs, and the job runs in a separate Go routine to avoid blocking the `ServeHTTP` hook.

> Also we need to make sure triggering the job manually cannot lead to having two jobs running.

The control flow of the API call now uses the same logic as the scheduled job, which checks for an existing runner before processing the legal holds. Are there further additions/modifications needed to coordinate this?